civihr_employee_portal/tdd/PublicFirewallTest.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+use Drupal\civihr_employee_portal\Security\PublicFirewall;
+
+// required as we have no autoloader inside this extension
+require_once __DIR__ . '/../src/Security/PublicFirewall.php';
+
+class PublicFirewallTest extends \PHPUnit_Framework_TestCase {
+
+  /**
+   * @test
+   */
+  public function willAllowAuthenticatedUser() {
+    $user = new \stdClass();
+    $user->uid = 132;
+
+    $firewall = new PublicFirewall();
+    $this->assertTrue($firewall->canAccess($user, '/foo'));
+  }
+
+  /**
+   * @test
+   * @dataProvider protectedRouteProvider
+   * @param $route
+   */
+  public function willDenyAnonymousUserAsExpected($route) {
+    $user = new \stdClass();
+    $user->uid = 0;
+    $firewall = new PublicFirewall();
+    $this->assertFalse($firewall->canAccess($user, $route));
+  }
+
+  /**
+   * @test
+   * @dataProvider publicRouteProvider
+   * @param $route
+   */
+  public function willAllowAnonymousUserAsExpected($route) {
```

Comment:
Sure, I just thought I'd chance my arm putting it in to see the reaction. We had a discussion about this where I worked before, I was pro using 'test' prefix first but they won me over :-)